### PR TITLE
"populate noise" step of chunk generation

### PR DESCRIPTION
**Pull request by kralverde**
This PR implements (most of) the "populate noise" stage of chunk generation

The blender (for smoothing (i think)) and the beardifier (for leaving space for world structures) are not implemented in this PR

**This PR changes the default world gen to this noise generation**

## Checklist
- [x] Re-implement chunk noise
- [x] Optimize chunk noise references
- [x] Look into hot code if chunks take too long to load
- [x] Add tests for actual chunk noise
- [x] Code is well-formatted and adheres to project style guidelines: `cargo fmt`
- [x] Code does not produce any clippy warnings `cargo clippy`

### CODE REVIEW

**Comment by kralverde** [comment]
The chunk noise generation and population each take like 0.5 seconds, so 1 second per chunk. I'll now start a deep dive into finding optimizations

**Comment by kralverde** [comment]
Chunk gen times are now ~0.1sec which i think is good enough for now

**Comment by OfficialKris** [comment]
This is a very large PR. I'm just curious, what was the methodology of this implementation?

**Comment by kralverde** [comment]
What do you mean by methodology?

**Comment by kralverde** [comment]
But basically I would look at the source code, write an equivalent in rust, and then write a bunch of unit tests for correctness. It was not fun lol

**Comment by kralverde** [comment]
If this PR is merged as-is, the world generation would become that of the populate noise

**Comment by kralverde** [comment]
Final bench is ~80msec per chunk on my machine

File: pumpkin-world/assets/no_bench_no_blend_0_0.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
+package com.example.mixin;
```
Comment (Snowiiii) [inline pumpkin-world/assets/no_bench_no_blend_0_0.java:1]: What the hell is this class and why is this in pumpkin-world ?

Comment (kralverde) [reply pumpkin-world/assets/no_bench_no_blend_0_0.java:1]: Its code to generate the no_bench_no_blend_0_0.chunk file

**Comment by kralverde** [comment]
~60msec/chunk on my machine with the unsafe code

**Comment by Q2297045667** [comment]
Report BUG
["populate noise" step of chunk generation](https://github.com/Snowiiii/Pumpkin/issues/340)

**Comment by kralverde** [comment]
> Report BUG
> 
> ["populate noise" step of chunk generation](https://github.com/Snowiiii/Pumpkin/issues/340)

I don't think this is necessarily relevant to this PR

**Comment by kralverde** [comment]
~50msec/chunk on my machine

**Comment by kralverde** [comment]
currently benching at ~55msec/chunk now

File: pumpkin-world/assets/no_bench_no_blend_0_0.java (hunk 1 above)
Comment (Snowiiii) [reply pumpkin-world/assets/no_bench_no_blend_0_0.java:1]: Why you don't put it into the extractor/ ?

Comment (kralverde) [reply pumpkin-world/assets/no_bench_no_blend_0_0.java:1]: I didn't think about that, I'll do that

File: extractor/src/main/kotlin/de/snowii/extractor/extractors/Tests.kt [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+				return result
+			}
+		}
+		System.err.println("No valid method found for block state sampler!");
+		return null;
+	}
+
+	override fun gson_builder(): Gson {
+		return GsonBuilder().create()
+	}
```
Comment (Snowiiii) [inline extractor/src/main/kotlin/de/snowii/extractor/extractors/Tests.kt:113]: Is the only reason your using this is that you don't want pretty printing ?. Feel free to remove pretty printing from everything

**Comment by kralverde** [comment]
@Snowiiii should be good to go now I think

**Comment by OfficialKris** [comment]
Runs great on my computer, very impressive. But why does it have more loc than Minecraft itself?

**Comment by kralverde** [comment]
> Runs great on my computer, very impressive. But why does it have more loc than Minecraft itself?

Loc?

**Comment by user622628252416** [comment]
> > Runs great on my computer, very impressive. But why does it have more loc than Minecraft itself?
> 
> Loc?

lines of code

**Comment by OfficialKris** [comment]
How are the asset files generated? Are they from minecraft? Is it possible to generate them at compile time? each of them are 3MB and that could be a problem to the git repo if they change often (like version updates).

**Comment by kralverde** [comment]
> How are the asset files generated? Are they from minecraft? Is it possible to generate them at compile time? each of them are 3MB and that could be a problem to the git repo if they change often (like version updates).

They're generated from Java, not really able to generate them thru rust... maybe I could write a script to generate and run the tests? But then they couldn't be run with the PR tests.

File: extractor/src/main/kotlin/de/snowii/extractor/extractors/Tests.kt (hunk 2 above)
Comment (kralverde) [reply extractor/src/main/kotlin/de/snowii/extractor/extractors/Tests.kt:113]: @Snowiiii removed

**Pull request merged**